contracts/src/test/TestContracts/InvariantsTestHandler.t.sol

```diff
@@ -1535,6 +1535,7 @@ contract InvariantsTestHandler is Assertions, BaseHandler, BaseMultiCollateralTe
         v.batchManager = _batchManagerOf[i][v.troveId];
         v.batchManagementFee = v.c.troveManager.getLatestBatchData(v.batchManager).accruedManagementFee;
         v.t = v.c.troveManager.getLatestTroveData(v.troveId);
+        vm.assume(v.t.entireDebt > 0); // skip troves if there’s no pending debt
```

Comment:
This goes against the spirit of the invariant tests. If we know an operation is expected to fail, we go ahead anyway, and make sure that it does fail, and for the right reason.

Reply:
Also, when an operation does succeed, we should ensure that all preconditions are met. We are adding a new precondition (debt > 0) which means a corresponding check should be added.